spec/tasks/one_off_spec.rb

```diff
@@ -63,4 +63,57 @@
       expect(maven2_version_dependency.platform).to eq("Maven")
     end
   end
+
+  describe "delete_ignored_maven_versions_and_resync_packages" do
+    let(:project) { create(:project, platform: "Maven", name: "project") }
+
+    context "with non-Maven project" do
+      let(:version) { create(:version, project: project, number: "1.0.0", repository_sources: nil) }
+
+      it "does not delete project or its version" do
+        Rake::Task["one_off:delete_ignored_maven_versions_and_resync_packages"].invoke("yes")
+
+        expect(Project.all).to match_array([project])
+        expect(Version.all).to match_array([version])
+      end
+    end
+
+    context "with Maven project with no ignored versions" do
+      let(:maven_source_version) { create(:version, project: project, number: "1.0.0", repository_sources: ["Maven"]) }
+      let(:google_source_version) { create(:version, project: project, number: "2.0.0", repository_sources: ["Google"]) }
+
+      it "does not delete project or its versions" do
+        Rake::Task["one_off:delete_ignored_maven_versions_and_resync_packages"].invoke("yes")
+
+        expect(Project.all).to match_array([project])
+        expect(Version.all).to match_array([maven_source_version, google_source_version])
+      end
+    end
+
+    context "with Maven project with some ignored versions" do
+      let(:maven_source_version) { create(:version, project: project, number: "1.0.0", repository_sources: ["Maven"]) }
+      let(:google_source_version) { create(:version, project: project, number: "2.0.0", repository_sources: ["Google"]) }
+      let(:no_source_version) { create(:version, project: project, number: "3.0.0", repository_sources: nil) }
+      let(:ignored_source_version) { create(:version, project: project, number: "4.0.0", repository_sources: ["Other"]) }
+
+      it "deletes ignored versions but not the project or its other versions" do
+        Rake::Task["one_off:delete_ignored_maven_versions_and_resync_packages"].invoke("yes")
+
+        expect(Project.all).to match_array([project])
+        expect(Version.all).to match_array([maven_source_version, google_source_version])
+      end
```

Comment:
One thing here, this is testing if these records get deleted by making sure they don't exist, but where do we ensure they ever existed? This test still passes even without running the task

Reply:
Hm good point. I usually don't add an expectation that spec setup worked correctly, and only add expectations for the actual behavior we're testing. But you're right, so maybe it's worth doing that for these specs.

Reply:
Alternatively could do some `let!` in the setup as well, though not sure I prefer that. Will investigate a good approach and fix either way, good catch.

Reply:
Can you use the `change` matcher?

Reply:
Wrapping it in a `change` expectation would be nice. Using `let!` should work fine as well. I wouldn't say you necessarily need to assert that they exist, just to realize that's the assumption being made (and to make sure to start with a test that fails)

Reply:
I made the relevant changes here, although if there's a better way to write the specs feel free to let me know. Not sure, for example, if there's a good way to confirm a `.not_change` without having to write expectations before calling the task, to confirm initial state.

This said, these specs won't hang around after the task is run, so it's not a huge deal to fix as long as we're confident it works.

Reply:
Thanks for including specs in the first place. The task itself looked correct I just wanted to back up and make sure we didn't commit a classic goof

For future ref you shooooould be able to write `expect { some_op }.to not_change(thing.prop1).from("some initial value")` -- ie just omit the `.to()` matcher.